pkg/trace/api/api.go

```diff
@@ -84,6 +84,10 @@ type HTTPReceiver struct {
 
 	// outOfCPUCounter is counter to throttle the out of cpu warning log
 	outOfCPUCounter *atomic.Uint32
+
+	// PreHookFunc is a function executed when a payload is received from a tracer
+	// used in serverless context to block until ready to properly handle the payload
+	PreHookFunc func()
```

Comment:
Let's call this maybe `onHandleTraces`.

Reply:
I capitalized it since it needs to be accessible from a serverless package